Modified scarab container

mkong0216 opened the pull request:
Changed how scarab displays its icons (github, facebook, etc.) to "float: left" so that we no longer have to calculate the left property for each icon and so that the scarab container resizes itself to fit all its icons. 

Also added data URI for info button

Commit:
Modified scarab 

Changed how scarab displays its icons (github, facebook, etc.) to "float: left" so that we no longer have to calculate the left property for each icon and so that the scarab container resizes itself to fit all its icons. 

Also added data URI for info button

Commit:
Update scarab.js

If github link not given by user, github icon no longer included in the scarab 
Responsive design implemented for description pop up box (will always match size of scarab container)

louh commented on src/scarab.js:
```diff
@@ -124,6 +149,7 @@ var MapzenScarab = (function () {
     linkEl.href = linkHref
     linkEl.target = '_blank'
     linkEl.className = 'mz-bug-' + id + '-link'
+    linkEl.classList.add('mz-bug-icons')
```
We can't use `classList` in very low-level libraries because it's not supported below IE10. We have to add it directly on the string value for the `.className` property (above line).

louh commented on src/scarab.js:
```diff
@@ -152,7 +178,33 @@ var MapzenScarab = (function () {
+  function _buildDescription(id) {
+    var infoBox = document.createElement('div')
+    infoBox.className = "mz-bug-" + id
+    infoBox.innerHTML = opts.description 
```
let's use `.textContent` instead of `.innerHTML`. The value set here is user-generated content and if we allow any HTML then you run the risk of allowing malicious / insecure HTML. `textContent` will be safer. We will address things like links separately and in the future...

louh commented on src/scarab.js:
```diff
@@ -171,12 +223,17 @@ var MapzenScarab = (function () {
     // Build links
     this.rebuildLinks()
-
+    
```
Remove trailing spaces

louh commented on src/scarab.css:
```diff
@@ -150,16 +167,29 @@
-  .mz-bug-facebook-link {
-    left: 96px;
-  }
+.mz-bug-info{
```
Space between class name and `{`

louh commented on src/scarab.js:
```diff
@@ -92,7 +92,7 @@ var MapzenScarab = (function () {
     var mapzenLink = opts.link || DEFAULT_LINK
     var mapzenTitle = (opts.name) ? opts.name + ' · Powered by Mapzen' : 'Powered by Mapzen'
-    var githubLink = opts.repo || DEFAULT_GITHUB_LINK
+    var githubLink = opts.repo 
```
Because `DEFAULT_GITHUB_LINK` isn't used anymore, remove it from line 14

louh replied:
Let's also stop using `githubLink` because we are reading `opts.repo` directly.

louh commented on src/scarab.js:
```diff
@@ -110,11 +110,36 @@ var MapzenScarab = (function () {
+    
+    // Creating github icon button if needed
+    if (opts.repo) {
+      var githubEl = _createButtonEl('github', githubLink, 'View source on GitHub', _onClickGitHub)
```
Use `opts.repo` instead of `githubLink` because they are the same.
Also, because `githubEl` is being created conditionally here, we should remove it from being created in line 107.

Commit:
Fixed requested changes

Commit:
Fixed requested changes

Commit:
Added new helper function

Added new function to help center scarab without having to change margin left in css stylesheet every time a new button is added or a button is taken away from scarab (not sure if this is needed, but thought it would be easier)

Commit:
Added helper function is scarab.js 

No longer need to style margin-left, did this in scarab.js 
Not sure if needed, but I found using helper function, centerScarab(), easier

Commit:
Got rid of querySelectors

Got rid of the querySelectors used to find the description box in resizeDescription and _onClickInfo

Commit:
Update scarab.js

Decreased number of times document.getElementById was used
